aineko/core/dataset.py

```diff
@@ -1,525 +1,276 @@
+            raise DatasetError(message) from e
+
+    def write(self, *args: T, **kwargs: T) -> Any:
+        """Write to the dataset via the query layer."""
+        try:
+            return self._write(*args, **kwargs)
+        except DatasetError:
+            raise
+        except Exception as e:
+            message = f"Failed to write dataset {self.name}."
+            raise DatasetError(message) from e
+
+    def create(self, *args: T, **kwargs: T) -> Any:
+        """Create the dataset storage layer."""
+        try:
+            return self._create(*args, **kwargs)
+        except DatasetError:
+            raise
+        except Exception as e:
+            message = f"Failed to create dataset {self.name}."
+            raise DatasetError(message) from e
+
+    def delete(self, *args: T, **kwargs: T) -> Any:
+        """Delete the dataset storage layer."""
+        try:
+            return self._delete(*args, **kwargs)
+        except DatasetError:
+            raise
+        except Exception as e:
+            message = f"Failed to delete dataset {self.name}."
+            raise DatasetError(message) from e
+
+    def exists(self, *args: T, **kwargs: T) -> Any:
+        """Check if the dataset storage layer exists."""
+        try:
+            return self._exists(*args, **kwargs)
+        except DatasetError:
+            raise
+        except Exception as e:
+            message = f"Failed to check if dataset {self.name} exists."
+            raise DatasetError(message) from e
+
+    def initialize(self, *args: T, **kwargs: T) -> Any:
+        """Initialize the dataset query layer."""
+        try:
+            return self._initialize(*args, **kwargs)
+        except DatasetError:
+            raise
+        except Exception as e:
+            message = f"Failed to initialize dataset {self.name}."
+            raise DatasetError(message) from e
+
+    @abc.abstractmethod
+    def _read(self, *args: T, **kwargs: T) -> Any:
+        """Subclass implementation to read the dataset."""
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def _write(self, *args: T, **kwargs: T) -> Any:
+        """Subclass implementation to write the dataset."""
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def _create(self, *args: T, **kwargs: T) -> Any:
+        """Subclass implementation to create the dataset."""
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def _delete(self, *args: T, **kwargs: T) -> Any:
+        """Subclass implementation to delete the dataset."""
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def _initialize(self, *args: T, **kwargs: T) -> Any:
+        """Subclass implementation to initialize the dataset query layer."""
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def _describe(self, *args: T, **kwargs: T) -> str:
+        """Subclass implementation to describe the dataset metadata."""
+        return f"Dataset name: {self.name}"
```

Comment:
This is supposed to be a common place to return metadata about a specific dataset. By default, it just returns the name, but particular implementations can add data to this. It may not be strictly necessary (and if it is going to be a string, we could just make it the __str__ or __repr__ method)

Reply:
Let's remove it and add a __str__ method instead. (just before the  `from_config` method.
I suggest:
```python
    def __str__(self) -> str:
        """Return the string representation of the dataset."""
        return f"{self.__class__.__name__}({self.name})"
```

Reply:
this makes more sense. updated this.